packages/model-viewer/src/three-components/CachingGLTFLoader.ts

```diff
@@ -178,17 +180,51 @@ export class CachingGLTFLoader {
         }
         node.castShadow = true;
         const mesh = node as Mesh;
+        let transparent = false;
         const materials =
             Array.isArray(mesh.material) ? mesh.material : [mesh.material];
         materials.forEach(material => {
           if ((material as any).isMeshStandardMaterial) {
+            if (material.transparent && material.side === DoubleSide) {
+              transparent = true;
+              material.side = FrontSide;
+            }
             this.roughnessMipmapper.generateMipmaps(
                 material as MeshStandardMaterial);
           }
         });
+
+        if (transparent) {
+          duplicate.push(mesh);
+        }
       });
     }
 
+    // We duplicate transparent, double-sided meshes and render the back face
+    // before the front face. This creates perfect triangle sorting for all
+    // convex meshes. Sorting artifacts can still appear when you can see
+    // through more than two layers of a given mesh, but this can usually be
+    // mitigated by the author splitting the mesh into mostly convex regions.
+    // The performance cost is not too great as the same shader is reused and
+    // the same number of fragments are processed; only the vertex shader is run
+    // twice.
+    for (let i = 0; i < duplicate.length; i++) {
```

Comment:
If the only thing you are using `i` for is to index into the array, consider rewriting as:

```typescript
for (const mesh of duplicate) {
  // ...
}
```